### PR TITLE
AF-2847: Adding template to the REST API /spaces/{spaceName}/project command in order to create new project based on the archetype template

### PR DESCRIPTION
**Pull request by akumar074**
**JIRA**: 
[BAPL-1900](https://issues.redhat.com/browse/BAPL-1900)
[RHPAM-3643](https://issues.redhat.com/browse/RHPAM-3643)
[AF-2847](https://issues.redhat.com/browse/AF-2847)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Comment by akumar074** [comment]
Jenkins retest this

File: kie-wb-common-screens/kie-wb-common-archetype-mgmt/kie-wb-common-archetype-mgmt-backend/src/main/java/org/kie/workbench/common/screens/archetype/mgmt/backend/service/ArchetypeServiceImpl.java [hunk 1]
```diff
@@ -374,6 +374,18 @@ private void checkTemplateShouldBeValid(final String alias) {
...
 
+    @Override
+    public Repository getTemplateRepository(String alias, String spaceName) {
+        final Map<String, Boolean> archetypeMap = archetypePreferencesManager.getArchetypeSelectionMap(spaceName);
+        if(!archetypeMap.containsKey(alias))
+            throw new IllegalArgumentException(
+                    String.format("Template repository %s is not available", alias));
+        if (!archetypeMap.get(alias))
+            throw new IllegalArgumentException(
+                    String.format("Template repository %s is not available for space %s", alias, spaceName));
```
[the remark this answers is not in the document]
Comment (akumar074) [reply kie-wb-common-screens/kie-wb-common-archetype-mgmt/kie-wb-common-archetype-mgmt-backend/src/main/java/org/kie/workbench/common/screens/archetype/mgmt/backend/service/ArchetypeServiceImpl.java]: Hi @caponetto , I agree to keep the `archetypeSelectionMap` in `ArchetypePreferenceManager`. I have made the required changes, please take a look.
Thanks!

**Comment by caponetto** [comment]
Jenkins retest this

**Comment by caponetto** [comment]
The failed test seems unrelated.
```
org.kie.workbench.common.screens.server.management.client.container.status.ContainerRemoteStatusPresenterTest.testOnServerInstanceUpdatedNewInstance
```

**Review by caponetto — APPROVED** [review]

**Pull request merged**